modules/worker_groups/main.tf

```diff
@@ -0,0 +1,241 @@
+resource "aws_autoscaling_group" "workers" {
+  for_each = local.worker_group_configurations
+
+  name_prefix = join(
+    "-",
+    compact(
+      [
+        var.cluster_name,
+        each.key,
+        each.value["asg_recreate_on_change"] ? random_pet.workers[each.key].id : ""
+      ]
+    )
+  )
+
+  desired_capacity        = each.value["asg_desired_capacity"]
+  max_size                = each.value["asg_max_size"]
+  min_size                = each.value["asg_min_size"]
+  force_delete            = each.value["asg_force_delete"]
+  target_group_arns       = each.value["target_group_arns"]
+  service_linked_role_arn = each.value["service_linked_role_arn"]
+  vpc_zone_identifier     = each.value["subnets"]
+  protect_from_scale_in   = each.value["protect_from_scale_in"]
+  suspended_processes     = each.value["suspended_processes"]
+
+  enabled_metrics = each.value["enabled_metrics"]
+
+  placement_group = each.value["placement_group"]
+
+  termination_policies      = each.value["termination_policies"]
+  max_instance_lifetime     = each.value["max_instance_lifetime"]
+  default_cooldown          = each.value["default_cooldown"]
+  health_check_grace_period = each.value["health_check_grace_period"]
+
+  dynamic mixed_instances_policy {
+    iterator = item
+    for_each = ((lookup(var.worker_groups[each.key], "override_instance_types", null) != null) || (lookup(var.worker_groups[each.key], "on_demand_allocation_strategy", null) != null)) ? list(each.value) : []
```

Comment:
does `var.worker_groups[each.key]` different from `each.value` ? If not, I'll prefer to use that. It'll more consistant for readability.

Reply:
Unfortunately, it is different according to my analysis. This is only to avoid changing old logic. See here: https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/workers_launch_template.tf#L93

Reply:
Sorry, respond too often.
This is what came as an argument from user: var.worker_groups_launch_template[count.index] while "each.value" represents what is already a merge with module defaults. Which IMHO could change original behaviour